tensorflow/lite/kernels/squeeze_test.cc

```diff
@@ -44,74 +43,112 @@ class BaseSqueezeOpModel : public SingleOpModel {
   int output_;
 };
 
-class FloatSqueezeOpModel : public BaseSqueezeOpModel {
+class SqueezeOpModel : public BaseSqueezeOpModel {
  public:
   using BaseSqueezeOpModel::BaseSqueezeOpModel;
 
-  void SetInput(std::initializer_list<float> data) {
+  template <typename T>
+  void SetInput(std::initializer_list<T> data) {
     PopulateTensor(input_, data);
   }
 
-  std::vector<float> GetOutput() { return ExtractVector<float>(output_); }
+  template <typename T>
+  std::vector<T> GetOutput() {
+    return ExtractVector<T>(output_);
+  }
   std::vector<int> GetOutputShape() { return GetTensorShape(output_); }
 };
 
-TEST(FloatSqueezeOpTest, SqueezeAll) {
-  std::initializer_list<float> data = {
-      1.0,  2.0,  3.0,  4.0,  5.0,  6.0,  7.0,  8.0,  9.0,  10.0, 11.0, 12.0,
-      13.0, 14.0, 15.0, 16.0, 17.0, 18.0, 19.0, 20.0, 21.0, 22.0, 23.0, 24.0};
-  FloatSqueezeOpModel m({TensorType_FLOAT32, {1, 24, 1}},
-                        {TensorType_FLOAT32, {24}}, {});
-  m.SetInput(data);
+template <typename T>
+void Check(std::initializer_list<int> input_shape,
```

Comment:
Minor suggestion: Can we rename this to `Test` to avoid confusion?
`CHECK` in Tensorflow codebase is usually an assertion.